### PR TITLE
Greg/travis

### PR DESCRIPTION
**Pull request by GregTheGreek**
## Changes
Adds support for:
- [x] Travis
- [x] CircleCI (cleans up file)
- [x] Adds `-short` to test command for memory intensive tests

## Tests:

To run the **short** test suite, where memory constraints exists in CI's, append the `-short` flag:
```
$ go test -short ./... 
```
When running the **full** test suite, omit the -short flag:
```
$ go test ./..
```

### Fixes: #60

### CODE REVIEW

**Comment by GregTheGreek** [comment]
@ansermino ready to rumble

File: codec/encode_test.go [hunk 1]
```diff
@@ -79,49 +79,65 @@ var encodeTests = []encodeTest{
 // Commented out testString3 since it causes the CI to fail.
```
Comment (noot) [inline codec/encode_test.go]: remove comment

Comment (ansermino) [reply codec/encode_test.go]: Please update but leave some note about failure

**Comment by dutterbutter** [comment]
Can we also change the ReadME badge to Travis and remove CircleCI

**Comment by GregTheGreek** [comment]
> Can we also change the ReadME badge to Travis and remove CircleCI

Yep! 1 minute

File: README.md [hunk 2]
```diff
@@ -23,6 +23,33 @@ go get -v -t -d ./...
 go test -v ./...
 ```
 
+#### Note on memory intensive tests
```
Comment (ansermino) [inline README.md]: Too much info for readme, please move to `CONTRIBUTING.md`

Comment (ansermino) [reply README.md]: (Perhaps leave an example like the existing on for `Test`)

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+- go get -v -t -d ./...
+
+before_script:
+  - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+  - chmod +x ./cc-test-reporter
+  - ./cc-test-reporter before-build
+
+script:
+- go test -v -short -coverprofile c.out ./... && ./cc-test-reporter after-build --exit-code $?
+- gofmt -d ./..
```
[the remark this answers is not in the document]
Comment (GregTheGreek) [reply .travis.yml]: Make commands aren't working

[the remark this answers is not in the document]
Comment (GregTheGreek) [reply .travis.yml]: I can re-write them, just won't be for a day or two

[the remark this answers is not in the document]
Comment (ansermino) [reply .travis.yml]: lets leave the makefile task open, we really need to just our CI back up at this point

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+language: go
+
+go:
+- 1.11
+
+install:
+# Install GolangCI-Linter
+- wget -O - -q https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s v1.15.0
+# Install all packages
+- go get -v -t -d ./...
```
[the remark this answers is not in the document]
Comment (ansermino) [reply .travis.yml]: Seems valid

[the remark this answers is not in the document]
Comment (noot) [reply .travis.yml]: does `-t` get the packages for the tests of dependencies or just the packages for our tests?

[the remark this answers is not in the document]
Comment (dutterbutter) [reply .travis.yml]: it gets the packages for the tests of dependencies I believe

**Pull request merged**